samples/snippets/src/main/java/com/example/spanner/RestoreBackupWithEncryptionKey.java

```diff
@@ -0,0 +1,102 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+// [START spanner_restore_backup_with_encryption_key]
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.spanner.BackupId;
+import com.google.cloud.spanner.Database;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Restore;
+import com.google.cloud.spanner.RestoreInfo;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.encryption.EncryptionConfigs;
+import com.google.spanner.admin.database.v1.RestoreDatabaseMetadata;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class RestoreBackupWithEncryptionKey {
+
+  static void restoreBackupWithEncryptionKey() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+    String backupId = "my-backup";
+    String kmsKeyName = "my-key";
+
+    try (Spanner spanner =
+        SpannerOptions.newBuilder().setProjectId(projectId).build().getService()) {
+      DatabaseAdminClient adminClient = spanner.getDatabaseAdminClient();
+      restoreBackupWithEncryptionKey(
+          adminClient,
+          projectId,
+          instanceId,
+          backupId,
+          databaseId,
+          kmsKeyName);
+    }
+  }
+
+  static Void restoreBackupWithEncryptionKey(DatabaseAdminClient adminClient,
+      String projectId, String instanceId, String backupId, String restoreId, String kmsKeyName) {
+    final Restore restore = adminClient
+        .newRestoreBuilder(
+            BackupId.of(projectId, instanceId, backupId),
+            DatabaseId.of(projectId, instanceId, restoreId))
+        .setEncryptionConfig(EncryptionConfigs.customerManagedEncryption(kmsKeyName))
+        .build();
+    final OperationFuture<Database, RestoreDatabaseMetadata> operation = adminClient
+        .restoreDatabase(restore);
+
+    Database database;
+    try {
+      System.out.println("Waiting for operation to complete...");
+      database = operation.get(1600, TimeUnit.SECONDS);
+    } catch (ExecutionException e) {
+      // If the operation failed during execution, expose the cause.
+      throw (SpannerException) e.getCause();
```

Comment:
It could in theory be a different exception than a `SpannerException`, so this would be safer:

```suggestion
      throw SpannerExceptionFactory.asSpannerException(e.getCause());
```

Reply:
Done.